src/langchain/llm-chat.ts

```diff
@@ -3,117 +3,181 @@ import {
   BaseChatModelParams,
 } from '@langchain/core/language_models/chat_models';
 import {
+  AIMessage,
+  AIMessageChunk,
   BaseMessage,
-  MessageType,
-  SystemMessage,
 } from '@langchain/core/messages';
 import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
-import { ChatResult } from '@langchain/core/outputs';
+import { ChatGenerationChunk, ChatResult } from '@langchain/core/outputs';
+import { BaseLanguageModelCallOptions } from '@langchain/core/language_models/base';
+import merge from 'lodash/merge.js';
 
-import { InvalidInputError } from '../errors.js';
-import { concatUnique } from '../helpers/common.js';
-import type { RequiredPartial } from '../helpers/types.js';
-import { TextGenerationCreateOutput } from '../schema.js';
-
-import { GenAIModel, GenAIModelOptions } from './llm.js';
+import { Client, Configuration } from '../client.js';
+import {
+  TextChatCreateInput,
+  TextChatCreateStreamInput,
+  TextGenerationCreateInput,
+} from '../schema.js';
+import { InternalError, InvalidInputError } from '../errors.js';
 
-export type RolesMapping = RequiredPartial<
-  Record<
-    MessageType,
-    {
-      stopSequence: string;
-    }
-  >,
-  'system'
->;
+export type GenAIChatModelParams = BaseChatModelParams &
+  Required<
+    Pick<TextGenerationCreateInput & TextChatCreateStreamInput, 'model_id'>
```

Comment:
Why is `TextGenerationCreateInput` used? 
I assume we want to use `TextChatCreateInput`.